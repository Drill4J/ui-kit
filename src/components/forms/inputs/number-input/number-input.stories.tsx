import React from 'react';
import { ComponentMeta } from '@storybook/react';
import 'twin.macro';

import { NumberInput } from './number-input';

export default {
  title: 'NumberInput',
  component: NumberInput,
} as ComponentMeta<typeof NumberInput>;

const Template = (args) => <NumberInput type="number" placeholder="enter text..." {...args} />;
export const Default = Template.bind({});
Default.args = { disabled: false };

export const Disabled = Template.bind({});
Disabled.args = { disabled: true };
