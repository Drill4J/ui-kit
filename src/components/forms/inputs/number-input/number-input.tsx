import { styled } from 'twin.macro';

import { Input } from '../input';
import { InputProps } from '../input-types';

export const NumberInput = ({ ...restProps }: InputProps) => <InputElement {...restProps} />;

const InputElement = styled(Input)`
  input {
    width: 60px;
    height: 40px;
    padding: 0 8px;

    -moz-appearance: textfield;

    ::-webkit-outer-spin-button,
    ::-webkit-inner-spin-button {
      -webkit-appearance: none;
    }
  }
`;
