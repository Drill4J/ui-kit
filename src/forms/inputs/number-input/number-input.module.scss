@import '../../../style-guide/index';

.number-input-wrapper {
  &__input-element {
    input {
      width: 60px;
      height: 32px;
      padding: 0 map-get($gutter-widths, extra-small);
      text-align: right;
    }
  }
}
