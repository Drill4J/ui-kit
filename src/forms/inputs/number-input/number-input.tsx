import styled from 'styled-components';

import { Input } from '../input';
import { InputProps } from '../input-types';

export const NumberInput = ({ className, ...restProps }: InputProps) => (
  <div className={className}>
    <InputElement {...restProps} />
  </div>
);

const InputElement = styled(Input)`
  input {
    width: 60px;
    height: 32px;
    padding: 0 8px;
    text-align: right;
  }
`;
