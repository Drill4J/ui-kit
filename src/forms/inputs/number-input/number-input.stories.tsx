import { storiesOf } from '@storybook/react';

import { NumberInput } from './number-input';

storiesOf('NumberInput', module).add('NumberInput types', () => (
  <>
    <h3>Simple input</h3>
    <NumberInput
      placeholder="enter text..."
    />
    <h3>Disabled input</h3>
    <NumberInput
      placeholder="enter text..."
      disabled
    />
    <h3>Error input</h3>
    <NumberInput
      placeholder="enter text..."
      error
    />
  </>
));
