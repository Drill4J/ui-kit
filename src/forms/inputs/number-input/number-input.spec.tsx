import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { NumberInput } from './number-input';

describe('NumberInput', () => {
  const mockFn = jest.fn();
  const props = {
    onChange: mockFn,
    disabled: false,
    checked: false,
    error: false,
    placeholder: 'placeholder initial state',
    value: 'input component',
    type: 'text',
  };

  it('Should render a NumberInput', () => {
    const tree = renderer
      .create(
        <>
          <NumberInput
            placeholder="enter text..."
          />
          <NumberInput
            placeholder="enter text..."
            disabled
          />
          <NumberInput
            placeholder="enter text..."
            error
          />
        </>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with initial state', () => {
    const wrapper = mount(
      <NumberInput {...props} />,
    );
    wrapper.find('input').simulate('change');
    expect(mockFn).toHaveBeenCalled();
    expect(wrapper.prop('value')).toContain('input component');
    expect(wrapper.prop('placeholder')).toContain('placeholder initial state');
  });
});
