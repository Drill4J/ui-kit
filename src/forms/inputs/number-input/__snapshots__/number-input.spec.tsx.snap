// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput Should render a NumberInput 1`] = `
Array [
  <div
    className="number-input-wrapper"
  >
    <div
      className="input-wrapper number-input-wrapper__input-element"
    >
      <input
        className="input-wrapper__input-element"
        placeholder="enter text..."
        value=""
      />
    </div>
  </div>,
  <div
    className="number-input-wrapper"
  >
    <div
      className="input-wrapper input-wrapper--disabled number-input-wrapper__input-element"
    >
      <input
        className="input-wrapper__input-element input-wrapper__input-element--disabled"
        disabled={true}
        placeholder="enter text..."
        value=""
      />
    </div>
  </div>,
  <div
    className="number-input-wrapper"
  >
    <div
      className="input-wrapper input-wrapper--error number-input-wrapper__input-element"
    >
      <input
        className="input-wrapper__input-element input-wrapper__input-element--error"
        placeholder="enter text..."
        value=""
      />
    </div>
  </div>,
]
`;
